### ISSUE TITLE
Resource leak?

### ISSUE THREAD

**Issue by zeshuai007** [bazaarvoice/jolt#912]
https://github.com/bazaarvoice/jolt/blob/63f62d460c08abe053d6bc5a7a6c56edda9dc023/complete/src/main/java/com/bazaarvoice/jolt/ChainrFactory.java#L92
Does FileInputStream need to be closed?
Isn't this a problem of resource leaks?

**Comment by KuKuDeCheng** [comment]
@zeshuai007  我跟了一下代码，后面通过try-resource关闭了parser
JsonUtils.java
```java
 public static Object jsonToObject( InputStream in ) {
        return util.jsonToObject( in );
    }
```
in JonUtilImpl.java
```java
@Override
    public Object jsonToObject( InputStream in ) {
        try {
            return objectMapper.readValue( in, Object.class );
        }
        catch ( IOException e ) {
            throw new JsonUnmarshalException("Unable to unmarshal JSON to an Object.", e );
        }
    }
```
And ,ObjectMapper.java
```java
@SuppressWarnings("unchecked")
    public <T> T readValue(InputStream src, Class<T> valueType)
        throws IOException, JsonParseException, JsonMappingException
    {
        return (T) _readMapAndClose(_jsonFactory.createParser(src), _typeFactory.constructType(valueType));
    } 
```
the _readMapAndClose method
![image](https://user-images.githubusercontent.com/21312425/70864162-7e1d0380-1f8a-11ea-9454-e62f87192756.png)

**Comment by zeshuai007** [comment]
@KuKuDeCheng 
Thanks, I quickly checked Jackson's ObjectMapper, and will help us close the input source by default, because AUTO_CLOSE_SOURCE defaults to true.
[](https://github.com/bazaarvoice/jolt/blob/63f62d460c08abe053d6bc5a7c6edda9dc023/cli/src/main/java/com/bazaarvoice/jolt/JoltCliUtilities.java#L54)
Therefore, it is not necessary to close again. If it is just a final check to ensure whether the input source is closed, you can put it in finally and judge isclosed. What do you think?

**Comment by KuKuDeCheng** [comment]
@zeshuai007 最后加资源检测关闭，个人感觉不太好，一是容易误导close的真正执行逻辑，二是没有必要的逻辑检测，就好像给人一种if(true=true)一样不舒服的感觉

**Comment by KuKuDeCheng** [comment]
@zeshuai007 有空看一下  https://nifichina.github.io/jolt/jolt详解.html   帮忙看下写的有没有问题，能交流的人还是比较少

**Comment by zeshuai007** [comment]
Yes, this is misleading, so you can delete it or use try-with-resources.
@KuKuDeCheng  Good! The documents you organize are great!
